### PR TITLE
CycloneDxReporter: Write out one BOM per project by default

### PR DESCRIPTION
**Pull request by sschuberth**
Now that reporters can output multiple files, write out one BOM per
project.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

### CODE REVIEW

File: reporter/src/main/kotlin/reporters/CycloneDxReporter.kt [hunk 1]
```diff
@@ -140,61 +115,116 @@ class CycloneDxReporter : Reporter {
...
+
+                bom.addExternalReference(ExternalReference.Type.WEBSITE, project.homepageUrl)
+
+                val licenseNames = project.declaredLicensesProcessed.allLicenses +
+                        input.ortResult.getDetectedLicensesForId(project.id, input.packageConfigurationProvider)
+
+                bom.addExternalReference(ExternalReference.Type.LICENSE, licenseNames.joinToString(", "))
+
+                bom.addExternalReference(ExternalReference.Type.BUILD_SYSTEM, project.id.type)
+                bom.addExternalReference(
```
Comment (fviernau) [inline reporter/src/main/kotlin/reporters/CycloneDxReporter.kt:153]: no empty line above this one seems inconsistent.

File: reporter/src/main/kotlin/reporters/CycloneDxReporter.kt [hunk 2]
```diff
@@ -140,61 +115,116 @@ class CycloneDxReporter : Reporter {
...
+                    ExternalReference.Type.OTHER,
+                    project.id.toPurl(),
+                    "Package-URL of the project"
+                )
+
+                val dependencies = project.collectDependencies()
+                val packages = input.ortResult.getPackages().mapNotNull { (pkg, _) ->
+                    pkg.takeIf { it.id in dependencies }
```
Comment (fviernau) [inline reporter/src/main/kotlin/reporters/CycloneDxReporter.kt:161]: Consider using filter instead.

File: reporter/src/main/kotlin/reporters/CycloneDxReporter.kt (hunk 1 above)
Comment (sschuberth) [reply reporter/src/main/kotlin/reporters/CycloneDxReporter.kt:153]: Right, I've changed the spacing to better group things that belong together.

File: reporter/src/main/kotlin/reporters/CycloneDxReporter.kt (hunk 2 above)
Comment (sschuberth) [reply reporter/src/main/kotlin/reporters/CycloneDxReporter.kt:161]: With `filter` I'd still need a separate `map` before or after, iterating over the packages two times in total. This way I only need to iterate once.

**Pull request merged**